services/webnn/ort/graph_builder_ort.cc

```diff
@@ -1506,6 +1506,78 @@ GraphBuilderOrt::AddLayerNormalizationOperation(
   return base::ok();
 }
 
+[[nodiscard]] base::expected<void, mojom::ErrorPtr>
+GraphBuilderOrt::AddLinearOperation(const mojom::Linear& linear) {
+  // Emulate a linear operation whose calculation follows the expression `alpha
+  // * x + beta`.
+  const mojom::Operand& input_operand = GetOperand(linear.input_operand_id);
+  const OperandDataType input_data_type = input_operand.descriptor.data_type();
+
+  std::string alpha_name;
+  std::vector<uint32_t> constant_dims = input_operand.descriptor.shape().empty()
+                                            ? std::vector<uint32_t>()
+                                            : std::vector<uint32_t>{1};
+  switch (input_data_type) {
+    case OperandDataType::kFloat16: {
+      std::vector<uint16_t> alpha_data_fp16(
+          1, fp16_ieee_from_fp32_value(linear.alpha));
+      ASSIGN_OR_RETURN(alpha_name, CreateInitializer<uint16_t>(
```

Comment:
Using `CreateScalarInitializer`?

Reply:
Done, I didn't realize that both Mul and Add operators support 0D broadcast operations🧐.
Please take another look, thanks a lot!